### PR TITLE
Refactor bundle sql

### PR DESCRIPTION
**Pull request by seallard**
# Summary
### Added
- Filter functions for bundles
- Tests

### Fixed
- Broken tests due to incompatible sqlalchemy version

### Changed
- Updated the workflow file for tests

## Review
- [x] code approved by VJ
- [x] tests executed by SA
- [x] "Merge and deploy" approved by VJ
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: housekeeper/store/api/find.py [hunk 1]
```diff
@@ -24,36 +26,49 @@ def bundles(self):
         LOG.info("Fetching all bundles")
         return self.Bundle.query
```
Comment (Vince-janv) [inline housekeeper/store/api/find.py:27]: We can use the new function here
```suggestion
        return _get_bundle_query()
```

File: housekeeper/store/api/find.py [hunk 2]
```diff
@@ -98,20 +113,20 @@ def files(
...
             except ValueError:
                 before_dt = get_date(before, "%Y-%m-%d %H:%M:%S")
-            query = query.join(models.Version).filter(models.Version.created_at < before_dt)
+            query = query.join(Version).filter(Version.created_at < before_dt)
 
         return query
 
     @staticmethod
-    def files_ondisk(file_objs: models.File) -> set:
+    def files_ondisk(file_objs: File) -> set:
```
Comment (Vince-janv) [inline housekeeper/store/api/find.py:129]:
```suggestion
    def files_ondisk(file_objs: File) -> Set[File]:
```

File: requirements.txt [hunk 3]
```diff
@@ -3,5 +3,5 @@ Click<7
 coloredlogs
 marshmallow
 pyyaml
-SQLAlchemy
+SQLAlchemy<1.4
```
Comment (Vince-janv) [inline requirements.txt:6]: Why is this necessary?

File: housekeeper/store/api/find.py [hunk 4]
```diff
@@ -24,36 +26,49 @@ def bundles(self):
...
 
-    def bundle(self, name: str = None, bundle_id: int = None) -> models.Bundle:
+    def _get_bundle_query(self) -> Query:
+        """Return bundle query."""
+        return self.Bundle.query
+
+    def bundle(self, name: str = None, bundle_id: int = None) -> Bundle:
         """Fetch a bundle from the store."""
         if bundle_id:
             LOG.info("Fetching bundle with id: %s", bundle_id)
```
Comment (Vince-janv) [inline housekeeper/store/api/find.py:36]: Could you refactor these to f-strings? 😊 
```suggestion
            LOG.info(f"Fetching bundle with id: {bundle_id}")
```

File: requirements.txt (hunk 3 above)
Comment (Vince-janv) [reply requirements.txt:6]: I see, good catch 👍

Comment (seallard) [reply requirements.txt:6]: When we are done with the refactoring, we should be able to use the latest version

**Comment by seallard** [comment]
Deployed to stage (modified prod script):
```bash
θ63° [sebastian.allard@hasta:~] [S_base] 3s $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh master
on the host hasta.scilifelab.se, continue
This will install the latest version of housekeeper on master in stage, are you sure? y/[n]y
/home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh: line 16: /home/proj/stage/bin/miniconda3/envs/master/bin/housekeeper: No such file or directory
Collecting git+https://github.com/clinical-genomics/housekeeper
  Cloning https://github.com/clinical-genomics/housekeeper to ./tmp/pip-req-build-glorc6i6
  Running command git clone --quiet https://github.com/clinical-genomics/housekeeper /home/sebastian.allard/tmp/pip-req-build-glorc6i6
  Resolved https://github.com/clinical-genomics/housekeeper to commit 8dde091f4195dc4c0023bda30b6a897143f4f5c5
  Preparing metadata (setup.py) ... done
Requirement already satisfied: Alchy in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (2.2.2)
Requirement already satisfied: Click<7 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (6.7)
Requirement already satisfied: coloredlogs in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (15.0.1)
Requirement already satisfied: marshmallow in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (3.19.0)
Requirement already satisfied: pyyaml in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (6.0)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (1.3.24)
Requirement already satisfied: rich in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (13.3.1)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Alchy->housekeeper==3.2.1) (6.0.2)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from coloredlogs->housekeeper==3.2.1) (10.0)
Requirement already satisfied: packaging>=17.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from marshmallow->housekeeper==3.2.1) (23.0)
Requirement already satisfied: markdown-it-py<3.0.0,>=2.1.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from rich->housekeeper==3.2.1) (2.2.0)
Requirement already satisfied: pygments<3.0.0,>=2.14.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from rich->housekeeper==3.2.1) (2.14.0)
Requirement already satisfied: mdurl~=0.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from markdown-it-py<3.0.0,>=2.1.0->rich->housekeeper==3.2.1) (0.1.2)
Building wheels for collected packages: housekeeper
  Building wheel for housekeeper (setup.py) ... done
  Created wheel for housekeeper: filename=housekeeper-3.2.1-py2.py3-none-any.whl size=24659 sha256=d5db1c5f11b8773c6e703aaef3395b7df6c4b5465c0d0217e9c2c82249ff8a7c
  Stored in directory: /home/sebastian.allard/tmp/pip-ephem-wheel-cache-o8m9i8i5/wheels/c4/bf/61/8d533b43ca0aff7ccf1c8bc69b86e5f9afda340643f5659fba
Successfully built housekeeper
Installing collected packages: housekeeper
  Attempting uninstall: housekeeper
    Found existing installation: housekeeper 3.2.0
    Uninstalling housekeeper-3.2.0:
      Successfully uninstalled housekeeper-3.2.0
Successfully installed housekeeper-3.2.1
/home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh: line 18: /home/proj/stage/bin/miniconda3/envs/master/bin/housekeeper: No such file or directory
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/sebastian.allard
done.
Log deploy... done.
/home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh: line 21: /home/proj/production/servers/resources/hasta.scilifelab.se/../test_version_command.sh: No such file or director
```

**Comment by seallard** [comment]
Deployed to prod:
```bash
θ63° [sebastian.allard@hasta:~] [S_base] 3s $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh master
on the host hasta.scilifelab.se, continue
This will install the latest version of housekeeper on master in stage, are you sure? y/[n]y
/home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh: line 16: /home/proj/stage/bin/miniconda3/envs/master/bin/housekeeper: No such file or directory
Collecting git+https://github.com/clinical-genomics/housekeeper
  Cloning https://github.com/clinical-genomics/housekeeper to ./tmp/pip-req-build-glorc6i6
  Running command git clone --quiet https://github.com/clinical-genomics/housekeeper /home/sebastian.allard/tmp/pip-req-build-glorc6i6
  Resolved https://github.com/clinical-genomics/housekeeper to commit 8dde091f4195dc4c0023bda30b6a897143f4f5c5
  Preparing metadata (setup.py) ... done
Requirement already satisfied: Alchy in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (2.2.2)
Requirement already satisfied: Click<7 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (6.7)
Requirement already satisfied: coloredlogs in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (15.0.1)
Requirement already satisfied: marshmallow in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (3.19.0)
Requirement already satisfied: pyyaml in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (6.0)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (1.3.24)
Requirement already satisfied: rich in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from housekeeper==3.2.1) (13.3.1)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Alchy->housekeeper==3.2.1) (6.0.2)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from coloredlogs->housekeeper==3.2.1) (10.0)
Requirement already satisfied: packaging>=17.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from marshmallow->housekeeper==3.2.1) (23.0)
Requirement already satisfied: markdown-it-py<3.0.0,>=2.1.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from rich->housekeeper==3.2.1) (2.2.0)
Requirement already satisfied: pygments<3.0.0,>=2.14.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from rich->housekeeper==3.2.1) (2.14.0)
Requirement already satisfied: mdurl~=0.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from markdown-it-py<3.0.0,>=2.1.0->rich->housekeeper==3.2.1) (0.1.2)
Building wheels for collected packages: housekeeper
  Building wheel for housekeeper (setup.py) ... done
  Created wheel for housekeeper: filename=housekeeper-3.2.1-py2.py3-none-any.whl size=24659 sha256=d5db1c5f11b8773c6e703aaef3395b7df6c4b5465c0d0217e9c2c82249ff8a7c
  Stored in directory: /home/sebastian.allard/tmp/pip-ephem-wheel-cache-o8m9i8i5/wheels/c4/bf/61/8d533b43ca0aff7ccf1c8bc69b86e5f9afda340643f5659fba
Successfully built housekeeper
Installing collected packages: housekeeper
  Attempting uninstall: housekeeper
    Found existing installation: housekeeper 3.2.0
    Uninstalling housekeeper-3.2.0:
      Successfully uninstalled housekeeper-3.2.0
Successfully installed housekeeper-3.2.1
/home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh: line 18: /home/proj/stage/bin/miniconda3/envs/master/bin/housekeeper: No such file or directory
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/sebastian.allard
done.
Log deploy... done.
/home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh: line 21: /home/proj/production/servers/resources/hasta.scilifelab.se/../test_version_command.sh: No such file or directory
[sebastian.allard@hasta:~] [S_base] 35s 1 $ down
[sebastian.allard@hasta:~] [base] 12s $ up
[sebastian.allard@hasta:~] [P_base] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-prod.sh
on the host hasta.scilifelab.se, continue
This will install the latest version of housekeeper on master in production, are you sure? y/[n]y
Collecting git+https://github.com/clinical-genomics/housekeeper
  Cloning https://github.com/clinical-genomics/housekeeper to ./tmp/pip-req-build-g5ez2o9a
  Running command git clone -q https://github.com/clinical-genomics/housekeeper /home/sebastian.allard/tmp/pip-req-build-g5ez2o9a
  Resolved https://github.com/clinical-genomics/housekeeper to commit 8dde091f4195dc4c0023bda30b6a897143f4f5c5
  Preparing metadata (setup.py) ... done
Requirement already satisfied: Alchy in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from housekeeper==3.2.1) (2.2.2)
Collecting Click<7
  Using cached click-6.7-py2.py3-none-any.whl (71 kB)
Requirement already satisfied: coloredlogs in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from housekeeper==3.2.1) (14.0)
Requirement already satisfied: marshmallow in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from housekeeper==3.2.1) (3.19.0)
Requirement already satisfied: pyyaml in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from housekeeper==3.2.1) (5.4)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from housekeeper==3.2.1) (1.3.24)
Requirement already satisfied: rich in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from housekeeper==3.2.1) (12.6.0)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from Alchy->housekeeper==3.2.1) (5.1.2)
Requirement already satisfied: humanfriendly>=7.1 in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from coloredlogs->housekeeper==3.2.1) (10.0)
Requirement already satisfied: packaging>=17.0 in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from marshmallow->housekeeper==3.2.1) (21.3)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from rich->housekeeper==3.2.1) (0.9.1)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from rich->housekeeper==3.2.1) (2.13.0)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /home/proj/production/bin/miniconda3/envs/P_base/lib/python3.10/site-packages (from packaging>=17.0->marshmallow->housekeeper==3.2.1) (3.0.9)
Building wheels for collected packages: housekeeper
  Building wheel for housekeeper (setup.py) ... done
  Created wheel for housekeeper: filename=housekeeper-3.2.1-py2.py3-none-any.whl size=24679 sha256=6f00aaf73d5eb02c5f0237146860c5890e10b476f988f4721990f091b6d5cc40
  Stored in directory: /home/sebastian.allard/tmp/pip-ephem-wheel-cache-ylwvp1ft/wheels/ba/10/8a/38c26617501f144820010b78c56daf2e5eac59f5f94a669d66
Successfully built housekeeper
Installing collected packages: Click, housekeeper
  Attempting uninstall: Click
    Found existing installation: click 7.1.2
    Uninstalling click-7.1.2:
      Successfully uninstalled click-7.1.2
  Attempting uninstall: housekeeper
    Found existing installation: housekeeper 3.2.0
    Uninstalling housekeeper-3.2.0:
      Successfully uninstalled housekeeper-3.2.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
cg 23.2.2 requires alive-progress, which is not installed.
loqusdb 2.6.10 requires click==7.1.2, but you have click 6.7 which is incompatible.
flask 2.2.2 requires click>=8.0, but you have click 6.7 which is incompatible.
flask-dance 5.1.0 requires Werkzeug<2.1, but you have werkzeug 2.2.2 which is incompatible.
cg 23.2.2 requires markupsafe<2.1, but you have markupsafe 2.1.2 which is incompatible.
cg 23.2.2 requires werkzeug<1.0.0, but you have werkzeug 2.2.2 which is incompatible.
Successfully installed Click-6.7 housekeeper-3.2.1
WARNING: You are using pip version 21.3.1; however, version 23.0.1 is available.
You should consider upgrading via the '/home/proj/production/bin/miniconda3/envs/P_base/bin/python3.10 -m pip install --upgrade pip' command.
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/sebastian.allard
done.
Log deploy... done.
/home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-prod.sh: line 21: /home/proj/production/servers/resources/hasta.scilifelab.se/../test_version_command.sh: No such file or directory
```
